src/raw/dockerapi/fetcher.go

```diff
@@ -0,0 +1,72 @@
+package dockerapi
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/docker/docker/api/types"
+	"github.com/newrelic/nri-docker/src/raw"
+)
+
+type Fetcher struct {
+	statsClient raw.DockerStatsClient
+}
+
+func NewFetcher(statsClient raw.DockerStatsClient) *Fetcher {
+	return &Fetcher{statsClient: statsClient}
+}
+
+func (f *Fetcher) Fetch(container types.ContainerJSON) (raw.Metrics, error) {
+	containerStats, err := f.ContainerStats(context.Background(), container.ID)
+	if err != nil {
+		return raw.Metrics{}, fmt.Errorf("could not fetch stats for container %s: %w", container.ID, err)
+	}
+	now := time.Now() // nolint: staticcheck
+	metrics := raw.Metrics{
+		Time:        now,
```

Comment:
Nit: we could just define it inline.
```suggestion
		Time:        time.Now(),
```

Reply:
I've updated it, although the `staticcheck` linter still complains if I remove the _nolint_:

```
src/raw/dockerapi/fetcher.go:27:16: SA4017: Now doesn't have side effects and its return value is ignored (staticcheck)
		Time:        time.Now(),
		             ^
```

🤔 I don't fully understand why it is complaining. I guess It could have something to do with the `Fetch` method not being called yet.